test/NuGet.Core.Tests/NuGet.Protocol.Tests/ServiceIndexResourceV3Tests.cs

```diff
@@ -24,6 +25,52 @@ public void Constructor_InitializesProperties()
             Assert.Equal("http://unit.test/b", resource.Entries[0].Uri.ToString());
         }
 
+        [Fact]
+        public void GetServiceEntries_WithResourceEndPoint_ThrowsException()
+        {
+            // Arrange
+            var serviceIndex = CreateServiceIndexWithHttpResources();
+            var resource = new ServiceIndexResourceV3(serviceIndex, DateTime.Now);
+
+            // Act & Assert
+            Assert.Throws<ArgumentException>(() => resource.GetServiceEntries("SearchQueryService"));
+            Assert.Throws<ArgumentException>(() => resource.GetServiceEntries("RegistrationsBaseUrl"));
+            Assert.Throws<ArgumentException>(() => resource.GetServiceEntries("LegacyGallery"));
+        }
+
+        [Fact]
+        public void GetServiceEntries_WithHttpResourceEndPointAndAllowInsecureConnections_Succeeds()
+        {
+            // Arrange
+            var serviceIndex = CreateServiceIndexWithHttpResources();
+            var resource = new ServiceIndexResourceV3(serviceIndex, DateTime.Now);
+            resource._allowInsecureConnections = true;
+
+            // Act
+            var searchRec = resource.GetServiceEntries("SearchQueryService").FirstOrDefault().Uri.ToString();
+            var regRec = resource.GetServiceEntries("RegistrationsBaseUrl").FirstOrDefault().Uri.ToString();
+            var legacyRec = resource.GetServiceEntries("LegacyGallery").FirstOrDefault().Uri.ToString();
+
+            // Assert
+            Assert.Equal(searchRec, "http://search/");
+            Assert.Equal(regRec, "http://reg/");
+            Assert.Equal(legacyRec, "http://legacy/");
+        }
+
+        [Fact]
+        public void GetServiceEntries_RequestsHttpsResourceInServiceIndexContainingOtherHttpResourcesWithoutAllowInsecureConnections_Succeeds()
+        {
+            // Arrange
+            var serviceIndex = CreateServiceIndexWithHttpResources();
+            var resource = new ServiceIndexResourceV3(serviceIndex, DateTime.Now);
+
+            // Act
+            var vulnRec = resource.GetServiceEntries("VulnerabilityInfo/6.7.0").FirstOrDefault().Uri.ToString();
```

Comment:
This abbreviation is not clear. Is it Vulnerable Resource, Vulnerability Resource, something else? Could we just use unabbreviated words for the var?